### DISCUSSION TITLE
[SOLVED] Unsafe member access Using "this" inside store actions (TypeScript)

### DISCUSSION THREAD

**Discussion by angelhdzmultimedia** [vuejs/pinia#962] [Help and Questions]
Using Pinia with TypeScript, I get an error when trying to read state items using "this" inside actions, and I have to tell the tslint to ignore it with` /* eslint-disable @typescript-eslint/no-unsafe-member-access */`. 

![image](https://user-images.githubusercontent.com/10191085/149606999-e7fd0193-e743-41ed-afc1-8bb9175a6613.png)

**Comment by angelhdzmultimedia** [comment]
### SOLVED?

I remembered that in TypeScript functions can have a reference to "this" as first parameter and add a type to that parameter, so this seems to solve my issue... 
![Code_WffqDmtdFp](https://user-images.githubusercontent.com/10191085/149646072-3f168423-1535-4af0-93b2-f38bb3c59d2c.png)

**Comment by posva** [comment]
This could be a misconfiguration in your tsconfig.json. A working config [can be seen here](https://github.com/piniajs/example-vue-2-vite/blob/main/tsconfig.json). Maybe you are modifying a default value that isn't there

**Reply by angelhdzmultimedia** [reply]
> This could be a misconfiguration in your tsconfig.json. A working config [can be seen here](https://github.com/piniajs/example-vue-2-vite/blob/main/tsconfig.json). Maybe you are modifying a default value that isn't there

I'll check the tsconfig, thanks. But I haven't touched anything, I created a fresh Quasar App and added pinia to it with a very basic store, the tsconfig is the one that Quasar creates.

**Reply by posva** [reply]
make sure you have an updated TS version then. 4.4 or 4.5. If not maybe people from quasar can help you on this one

**Comment by angelhdzmultimedia** [comment]
Still error after adding those new entries in the tsconfig and reloading VSCode.

![image](https://user-images.githubusercontent.com/10191085/149657671-2d7aa24b-01e1-40b7-912b-710666256f28.png)

tsconfig.json:

```
{
  "extends": "@quasar/app/tsconfig-preset",
  "compilerOptions": {
    "baseUrl": ".",
    "paths": {
      "@store/*":["./src/store/*"],
      "@components/*":["./src/components/*"],
      "@src/*":["./src/*"],
      "@router/*":["./src/router/*"],
      "@pages/*":["./src/pages/*"],
    },
    "target": "esnext",
    "module": "esnext",
    "moduleResolution": "node",
    "strict": true,
    "jsx": "preserve",
    "sourceMap": true,
    "resolveJsonModule": true,
    "esModuleInterop": true,
    "lib": ["esnext", "dom"]
  },
  "include": ["src/**/*.ts", "src/**/*.d.ts", "src/**/*.tsx", "src/**/*.vue"]
}

```

**Comment by angelhdzmultimedia** [comment]
Dependencies:
```

"dependencies": {
    "@quasar/extras": "^1.0.0",
    "axios": "^0.21.1",
    "core-js": "^3.6.5",
    "quasar": "^2.0.0"
  },
  "devDependencies": {
    "@babel/eslint-parser": "^7.13.14",
    "@quasar/app": "^3.0.0",
    "@types/node": "^12.20.21",
    "@typescript-eslint/eslint-plugin": "^4.16.1",
    "@typescript-eslint/parser": "^4.16.1",
    "eslint": "^7.14.0",
    "eslint-config-prettier": "^8.1.0",
    "eslint-plugin-vue": "^7.0.0"
  }
```

**Comment by angelhdzmultimedia** [comment]
### Solved

When I installed pinia `yarn add pinia`, it didn't get added in the package.json, so I guess it didn't get installed, or an old version got installed instead.
I hardcoded the latest version available and all issues are gone. Thanks everyone who cared.
